pkg/apimachinery/patch/patch.go

```diff
@@ -38,9 +37,73 @@ const (
 	PatchRemoveOp  = "remove"
 )
 
+type PatchSet interface {
```

Comment:
Interfaces are usually expressing behavior and not a set of methods for an object.
In Go, it is consider a good-practice to define them in the place they are used, and not in the package that defines their concrete.

Sometime, the solution is to have a dedicated package for interfaces, but IMO it should be used only in rare cases. The Go std have some examples of this.
But even there, there are multiple interfaces defined, not all-in-one.

Reply:
I am also missing the unit tests for this new object, expressing how it is expected to be used and showing basic scenarios.